.eslintrc.js

```diff
@@ -20,6 +20,7 @@ module.exports = {
   "rules": {
     "require-jsdoc": 0, // require JSDoc comments,
     "comma-dangle": 0, // enforce trailing commas,
-    "arrow-parens": 0 // enforce parens arround single arg arrow fcts
+    "arrow-parens": 0, // enforce parens arround single arg arrow fcts
+    "max-len": 0 // newsflash: computers have more than 80 columns now
```

Comment:
:man_shrugging: It's default in almost every linter still :slightly_frowning_face:

Reply:
At a few levels of indentation (easy to reach if you're writing things like classes) this makes your lines comically short.  I'm conceptually OK with a column limit but 80 is too short IMHO.

Reply:
I've always felt the same, but have just left it as-is, I wonder if there is a general agreement in the OS community on the subject :-).

Reply:
There are about as many arguments on column width as there are tabs vs spaces or brace style.

Personally for OSS projects that are checked out by a lot of different people, I think it makes some sense to settle on something that's readable on any type of modern computer with any set of eyeballs, and 80 columns is decent for that.  For projects with a small set of users who are using more advanced tools like IDEs and high-DPI monitors, I think 80 just ends up meaning half of everyone's IDE or editor is blank space, and people have to spend time regularly figuring out how to cram 81-characters of stuff into two lines to make the linter happy for no good reason.

Reply:
Well, I'll cast a vote for 80. For me it's less about screen size and more about legibility, e.g. https://baymard.com/blog/line-length-readability. Also, if I'm indenting enough that I'm regularly bumping up against my ruler, I take that as a heads-up that I may need to refactor -- see where I can de-nest conditionals, pull out functions, etc.

Reply:
It's worth mentioning that that article is about prose, not code, and in that case I totally agree.  Also wrapping a blog post or news article when it exceeds a particular line length is trivially easy and automatically handled by most tools.  If you have a four-line paragraph and you want to change one word in the middle, you don't have to sit down and figure out how to re-break the string into new chunks that don't exceed the line length.  But with code you do have to do that work.

Also, you don't have indentation and nesting levels and such like you do with code, so often you hit a situation where you are very nested (maybe something like an inner class) and only have 20 characters or so to play with per line.  That's not really in the spirit of what the article is saying, which is more about keeping information density per line low...and that is something I'm all for.  But I don't think 80 column limits really do that.  What they definitely do is handcuff the developer and make you consider line length to get the linter to be quiet, instead of things like low cyclomatic complexity and testability and readability being the forefront priorities.

If we really feel like 80 is important then I'll make it work, but I think I'm much happier with maybe a 120 or 160 column limit and a general team practice of keeping lines of code as succinct as possible and not jamming a bunch of cleverness in one line just because it's possible.

Reply:
I don't think I feel as strongly about it as you do, just casting my vote. And I agree with binding the hands of developers; I actually didn't start using a linter until my massive PR I filed on Friday afternoon, so I'm used to having the ability to break the 80 char rule when something truly merits it. :) All that said, with 2-space indentations, I wouldn't want to go far beyond 80 characters even with nesting factored in. In my mind nesting levels are part of readability; you have to remember which subcase of which subcase you're reading.

Reply:
Fair point.  I really mostly hit it when I have long strings.